In [45]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm     
from scipy.stats import chi2_contingency

In [20]:
# assignment no.3 hypothesis testing, q:- 01 
Cutlets = pd.read_csv("D:\Data Science Material\Assignments\Math Assignment 3 (Hypothesis)\Cutlets.csv")   

In [21]:
Cutlets.head()      # assignment no.3 hypothesis testing, q:- 01

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
Cutlets = Cutlets.rename(columns={'Unit A':'Unit_A','Unit B':'Unit_B'})    # assignment no.3 hypothesis testing,  q:- 01

In [5]:
Cutlets.head()   # assignment no.3 hypothesis testing, q:- 01

Unit_A  Unit_B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [6]:
Unit_A = Cutlets.Unit_A       # assignment no.3 hypothesis testing, q:- 01

In [7]:
Unit_A.head()            # assignment no.3 hypothesis testing,  q:- 01

0    6.8090
1    6.4376
2    6.9157
3    7.3012
4    7.4488
Name: Unit_A, dtype: float64

In [8]:
Unit_B = Cutlets.Unit_B         # assignment no.3 hypothesis testing,  q:- 01

In [9]:
Unit_B.head()               # assignment no.3 hypothesis testing,  q:- 01

0    6.7703
1    7.5093
2    6.7300
3    6.7878
4    7.1522
Name: Unit_B, dtype: float64

In [10]:
 # ind = independent samples
pvalues = stats.ttest_ind(Unit_A,Unit_B)        # assignment no.3 hypothesis testing, q:- 01

In [11]:
pvalues        # assignment no.3 hypothesis testing,  q:- 01

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

(At 5% significance level)
as pvalue=0.4722394724599501 > alfa = 0.05 accept null Hypothesis        

In [22]:
# assignment no.3 hypothesis testing, q:- 02
LabTAT = pd.read_csv("D:\Data Science Material\Assignments\Math Assignment 3 (Hypothesis)\LabTAT.csv")    

In [23]:
LabTAT.head()    # assignment no.3 hypothesis testing, q:- 02

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [14]:
LabTAT = LabTAT.rename(columns={'Laboratory 1':'Laboratory_1','Laboratory 2':'Laboratory_2','Laboratory 3':'Laboratory_3','Laboratory 4':'Laboratory_4'})
# assignment no.3 hypothesis testing, q:- 02

In [15]:
LabTAT.head()   # assignment no.3 hypothesis testing, q:- 02

Laboratory_1  Laboratory_2  Laboratory_3  Laboratory_4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [16]:
LabTAT.describe()     # assignment no.3 hypothesis testing, q:- 02

Laboratory_1  Laboratory_2  Laboratory_3  Laboratory_4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [18]:
p_value=stats.f_oneway(LabTAT.iloc[:,0],LabTAT.iloc[:,1],LabTAT.iloc[:,2],LabTAT.iloc[:,3])  # assignment no.3 hypothesis testing, q:- 02

In [19]:
p_value  # assignment no.3 hypothesis testing, q:- 02

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

(pvalue=2.1156708949992414e-57)> alfa 0.05  ie reject null hypothesis

In [28]:
# assignment no.3 hypothesis testing, q:- 03
BuyerRatio = pd.read_csv("D:\Data Science Material\Assignments\Math Assignment 3 (Hypothesis)\BuyerRatio.csv")

In [29]:
BuyerRatio   # assignment no.3 hypothesis testing, q:- 03

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [30]:
# assignment no.3 hypothesis testing, q:- 03
pvales = stats.f_oneway(BuyerRatio.iloc[:,1],BuyerRatio.iloc[:,2],BuyerRatio.iloc[:,3],BuyerRatio.iloc[:,4])

In [31]:
pvales  # assignment no.3 hypothesis testing, q:- 03

F_onewayResult(statistic=0.3068565327605033, pvalue=0.8204060154887464)

pvalue=0.8204060154887464  we accept null hypothesis

In [32]:
# assignment no.3 hypothesis testing, q:- 04
CostomerOrderForm = pd.read_csv("D:\Data Science Material\Assignments\Math Assignment 3 (Hypothesis)\CostomerOrderForm.csv")

In [34]:
CostomerOrderForm.head()   # assignment no.3 hypothesis testing, q:- 04

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [39]:
# assignment no.3 hypothesis testing, q:- 04
print("Counts of Phillippines \n",CostomerOrderForm.Phillippines.value_counts(),'\n')
print("Counts of Indonesia \n",CostomerOrderForm.Indonesia.value_counts(),'\n')
print("Counts of Malta \n",CostomerOrderForm.Malta.value_counts(),'\n')
print("Counts of India \n",CostomerOrderForm.India.value_counts(),'\n')

Counts of Phillippines 
 Error Free    271
Defective      29
Name: Phillippines, dtype: int64 

Counts of Indonesia 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 

Counts of Malta 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 

Counts of India 
 Error Free    280
Defective      20
Name: India, dtype: int64 



In [43]:
# array form    # assignment no.3 hypothesis testing, q:- 04
cust_order = np.array([[271,267,269,280],[29,33,31,20]])
cust_order

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

Assume Null Hypothesis as Ho: 
If independence of categorical variables (customer order forms defective %  does not varies by centre)

Thus, Alternative hypothesis as Ha
Dependence of categorical variables (customer order forms defective %  varies by centre)

In [46]:
# Chi2 contengency independence test   # assignment no.3 hypothesis testing, q:- 04
chi2_contingency(cust_order) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

(pvalue = 0.2771020991233135)  > alpha (0.05) ie . accept null hypothesis